src/SearchImpl/Indexing/CommitManager.cs

```diff
@@ -119,4 +121,58 @@ private void CommitWorker()
         }
 
     }
+
+    internal class NearRealTimeCommitManager : ICommitManager
+    {
+        private static readonly TimeSpan MaxWaitTime = TimeSpan.FromSeconds(10);
+        private DateTime _lastCommitTime;
+        private int _uncommittedActivityCount;
+
+        private static System.Timers.Timer _timer;
+        private static readonly int HearthBeatMilliseconds = 1000;
```

Comment:
There should not be a separate HeartBeatMilliseconds constant in every class that needs it.

Reply:
Why? Different functionality different hearbeat.